pressbooks/pressbooks-multi-institution
# feat: add institution column & filter for users in pressbooks-network-analytics

Issue: https://github.com/pressbooks/pressbooks-multi-institution/issues/41

Related PR (it needs to be in place for exploratory testing): https://github.com/pressbooks/pressbooks-network-analytics/pull/331

This PR hooks into 2 filters in Pressbooks Network Analytics defined in:
- https://github.com/pressbooks/pressbooks-network-analytics/pull/331/files#diff-5fd208ae68a4b262fe59b9cf5b480052bbfcf5507fc1be29c6778ba0b913b5fdR88
- https://github.com/pressbooks/pressbooks-network-analytics/pull/331/files#diff-ead8c28b91ca9599d2ec761826f8a8d5141cfa06d063ada0612f45cdad0f3584R124

This is to send the new `Institution` column and its data to the Users list in PB Analytics, as well as allow to download CSV with the new column and add a filter for super admins.

On the other hand, it fixes a bug. Now super admins and network managers cannot pick as an Insitutional Manager a user belonging to another institution. See: 
[Untitled_ Feb 28 2024 8_29 AM.webm](https://github.com/pressbooks/pressbooks-multi-institution/assets/13248424/1293c3c4-1c09-40ce-9e56-efc374fc6a95)


### Testing case
As a super admin or Network Manager:
- Go to the PB Analytics users list
- A new column "Institution" should be present after the Email column when this plugin is active.
- Users without any institution should be indicated as `Unassigned` in the institution column.
- you can order the users list by the new institution column.
- You should be able to see and use a new `Institution` filter.
- Download the CSV. The new institution column should be present in the downloaded file.

Bug fix:
- Go to the institution list and edit one of them.
- In the Institutional Managers selector you should not be able to pick another user that belongs to a different institution.

As an institutional manager:
- Go to PB Analytics users list
- You should be able to see only the users that belong to the institution for the institutional manager.
- A new column "Institution" should be present after the Email column when this plugin is active. The column should be filled with the institution name that belongs to the institutional manager logged in.
- Downloading the CSV list should contain the institution column.
